app/views/thredded/users/_recent_posts.html.erb

```diff
@@ -0,0 +1,20 @@
+<%#
+  This partial can be called from the parent app to render the user's recent posts.
+  The parent view context that renders this partial must include Thredded::ApplicationHelper.
+%>
+
+<%# @param user [Thredded.user_class] %>
+<%# @param limit [Fixnum] %>
+<%
+  # The thredded_current_user method is not available here when rendering from main_app.
+  current_user = send(Thredded.current_user_method) || Thredded::NullUser.new
+  posts = Thredded::PostsPageView.new(
+      current_user,
+      Pundit.policy_scope!(current_user, user.thredded_posts)
+          .where(messageboard_id: Pundit.policy_scope!(current_user, Thredded::Messageboard.all).pluck(:id))
+          .order_newest_first
+          .includes(:postable)
+          .limit(local_assigns[:limit] || 10))
+%>
```

Comment:
Was putting all of the above code in the view as a means to provide the person/team implementing this in their parent app with the code they could use in a controller? If that's the intent I would call that out in the comment above. At the risk of making poor assumptions and stating the obvious - all this work in the view is an anti-pattern 😜